src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioProjectSnapshotManagerDispatcher.cs

```diff
@@ -12,9 +14,14 @@ internal class VisualStudioProjectSnapshotManagerDispatcher : ProjectSnapshotMan
     {
         private const string ThreadName = "Razor." + nameof(VisualStudioProjectSnapshotManagerDispatcher);
 
-        public VisualStudioProjectSnapshotManagerDispatcher() : base(ThreadName)
-        {
+        private readonly ErrorReporter _errorReporter;
 
+        [ImportingConstructor]
+        public VisualStudioProjectSnapshotManagerDispatcher(SVsServiceProvider serviceProvider) : base(ThreadName)
+        {
+            _errorReporter = new VisualStudioErrorReporter(serviceProvider);
```

Comment:
Hmm, is `VisualStudioErrorReporter` something we should be exporting so we have a clear mechanism to log errors in MEF components?